PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-azure-sdk/src/main/java/com/microsoft/azure/toolkit/intellij/azure/sdk/buildtool/ConnectionStringCheck.java

```diff
@@ -0,0 +1,44 @@
+package com.microsoft.azure.toolkit.intellij.azure.sdk.buildtool;
+
+import com.intellij.codeInspection.*;
+import com.intellij.psi.*;
+import com.microsoft.applicationinsights.TelemetryClient;
+import org.jetbrains.annotations.NotNull;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * This class extends the LocalInspectionTool and checks for the usage of connection strings in the code.
+ * Connection strings that start with "DefaultEndpointsProtocol=http;AccountName=" are flagged as problems.
```

Comment:
Don't think this is the correct expectation of the rule. We should not be looking for the exact text represenatation i.e  "DefaultEndpointsProtocol=http;AccountName=" but the usage in APIs, if connection string APIs are used in user code.

Reply:
refactored it to check for use of connectionString API call, so no need to track presence of a connection string anymore:) this works for all services